docs/source/blocks/examples/custom-view-variations-and-schema-enhancer.md

```diff
@@ -0,0 +1,280 @@
+      },
+      title: {
+        title: intl.formatMessage(messages.title),
+      },
+    },
+    required: [],
+  };
+};
+
+export default Schema;
+```
+
+## Messages
+
+As you may have noted, you prepared the block for internationalization.
+{term}`Internationalization` (i18n) is the process of creating user interfaces which are suitable for different languages and cultural contexts.
+
+Create a file {file}`messages.js` in the same {file}`ExampleBlock06` folder with the following contents.
+
+```js
+import { defineMessages } from 'react-intl';
+
+const messages = defineMessages({
+  block06: {
+    id: 'block06',
+    defaultMessage: 'Block 06',
+  },
+  default: {
+    id: 'default',
+    defaultMessage: 'Default',
+  },
+  URL: {
+    id: 'URL',
+    defaultMessage: 'URL',
+  },
+  title: {
+    id: 'title',
+    defaultMessage: 'Title',
+  },
+});
+
+export default messages;
+```
+
+## View component
+
+For variations, the view component needs to use the variation template to render the contents of the block.
+
+You can do so using the `variation` from the `props` of the block.
+
+Create a file {file}`View.jsx` in the {file}`ExampleBlock06` folder with the following contents.
+
+```jsx
+import withBlockExtensions from '@plone/volto/helpers/Extensions/withBlockExtensions';
+import cx from 'classnames';
+import React from 'react';
+
+const View = (props) => {
+  // `data` holds the values entered in the block edit form.
+  // `className` holds the CSS class names injected to this block by Volto's `styleClassNameExtenders`.
+  // `style` holds the CSS properties injected into this block by Volto's `Block Style Wrapper`.
+  // `variation` holds the variation selected in the block editor, and it is an object as defined in the block configuration.
+  const { data, className, style, variation } = props;
+
+  const BodyTemplate = variation?.template;
+
+  return (
+    <div
+      className={cx(
+        'block',
+        'block06',
+        `block06-variation-${variation?.id}`,
+        className,
+      )}
+      style={style}
+    >
+      <BodyTemplate data={data} />
+    </div>
+  );
+};
+
+export default withBlockExtensions(View);
+```
+
+The `withBlockExtensions` {term}`HOC` makes the variation selector available in the block edit form and provides the `variation` property in the props.
+
+## Schema enhancer
+
+Next you need to configure the schema enhancer function.
+In this example, you will add a new field named `color` when using `schemaEnhancerVariation02`.
+
+Create a file {file}`enhancers.js` in the {file}`BlockSchema06` folder with the following content:
+
+```js
+const schemaEnhancerVariation02 = ({ formData, schema, intl }) => {
+  // schema holds the original schema (see the Schema.js file)
+  // so you need to define the new property under `schema.properties`
+  // and push its name to the relevant fieldset, in this case the first one (note the `fieldsets[0]`)
+  schema.properties.color = {
+    title: 'Color',
```

Comment:
@erral Since you are teaching the users to use the Intl message format, I think it would be worth it to use it also in the schema enhancer example. Therefore, we always teach them to create add-ons the proper way, thinking of internationalization for other users as well.